inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -113,16 +112,29 @@ def _handle_app_source_command(
         return Result(status=200, message=serialized_list)
 
     if "remove" in app_action:
-        keyname = parsed_head.get_children("applicationSource/remove/gpg")["keyname"]
+        keyname = None
+        try:
+            keyname = parsed_head.get_children("applicationSource/remove/gpg")["keyname"]
+        except XmlException:
+            # These children may not be present
+            logger.info(f"Optional GPG key parameters not present in manifest")
         filename = parsed_head.get_children("applicationSource/remove/repo")["filename"]
         application_source_manager.remove(
             ApplicationRemoveSourceParameters(file_name=filename, gpg_key_name=keyname)
         )
         return Result(status=200, message="SUCCESS")
 
     if "add" in app_action:
-        gpg_key_uri = parsed_head.get_children("applicationSource/add/gpg")["uri"]
-        gpg_key_name = parsed_head.get_children("applicationSource/add/gpg")["keyname"]
+        gpg_key_uri = None
+        gpg_key_name = None
+
+        try:
+            gpg_key_uri = parsed_head.get_children("applicationSource/add/gpg")["uri"]
+            gpg_key_name = parsed_head.get_children("applicationSource/add/gpg")["keyname"]
+        except XmlException:
+            # These children may not be present
+            logger.info(f"Optional GPG key parameters not present in manifest")
```

Comment:
**Suggestion:** Refactor the repeated try-except blocks for GPG key parameters into a separate function to improve code maintainability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        def get_optional_gpg_keys(parsed_head: XmlHandler, action: str) -> tuple:
            gpg_key_uri = None
            gpg_key_name = None
            try:
                gpg_keys = parsed_head.get_children(f"applicationSource/{action}/gpg")
                gpg_key_uri = gpg_keys.get("uri")
                gpg_key_name = gpg_keys.get("keyname")
            except XmlException:
                logger.info(f"Optional GPG key parameters not present in manifest for {action}")
            return gpg_key_uri, gpg_key_name

        # Usage in "remove" action
        keyname = get_optional_gpg_keys(parsed_head, "remove")[1]

        # Usage in "add" action
        gpg_key_uri, gpg_key_name = get_optional_gpg_keys(parsed_head, "add")
```